src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -1,56 +1,118 @@
 import { h, Component} from 'preact'
+import classNames from 'classnames'
+import io from 'socket.io-client'
+
 import theme from '../../Theme/style.css'
 import style from './style.css'
-import classNames from 'classnames'
+import Spinner from '../../Spinner'
 import { trackComponent } from '../../../Tracker'
+import { versionToBase36 } from '../../utils/versionMap'
 
 class CrossDeviceLink extends Component {
   constructor(props) {
-    super(props);
-    this.state = { copySuccess: false }
+    super(props)
+    this.state = {roomId: null}
+
+    if (!props.socket) {
+      const socket = io(process.env.DESKTOP_SYNC_URL)
+      props.actions.setSocket(socket)
+      this.listen(socket)
```

Comment:
so this is assuming the socket was set. probably better to set the listener under `propsWillChange`

Reply:
I don't quite folllow. do you mean `componentWillReceiveProps()` ?

Reply:
yes, `componentWillReceiveProps`